docs/models/google.md

```diff
@@ -108,6 +108,39 @@ agent = Agent(model)
 ...
 ```
 
+#### Customizing Model
+
+You can access models from the Model Garden, including but not limited to Gemini, available under your GCP project using one of the following `model_name` patterns: `{model_id}`, `publishers/{publisher}/models/{model_id}`, or `projects/{project}/locations/{location}/publishers/{publisher}/models/{model_id}`.
+
+```python
+from google.oauth2 import service_account
+from google.genai import Client
+
+from pydantic_ai import Agent
+from pydantic_ai.models.google import GoogleModel
+from pydantic_ai.providers.google import GoogleProvider
+
+location = 'your-region'
+project = 'your-project-id'
+credentials = service_account.Credentials.from_service_account_file(
+    'path/to/service-account.json',
+    scopes=['https://www.googleapis.com/auth/cloud-platform'],
+)
+client = Client(
+    vertexai=True,
+    location=location,
+    project=project,
+    credentials=credentials
+)
```

Comment:
Is a custom client needed in this case?

We should be able to use `GoogleProvider(credentials=credentials, project=project, location=location, vertexai=True)`

Reply:
No, just tried `GoogleProvider(credentials=credentials, project=project, location=location)`. Will update.